libminifi/include/core/PropertyDefinition.h

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <array>
+#include <optional>
+#include <string_view>
+#include <utility>
+
+#include "core/PropertyType.h"
+#include "utils/gsl.h"
+
+namespace org::apache::nifi::minifi::core {
+
+template<size_t NumAllowedValues = 0, size_t NumAllowedTypes = 0, size_t NumDependentProperties = 0, size_t NumExclusiveOfProperties = 0>
+struct PropertyDefinition {
+  std::string_view name;
+  std::string_view display_name;
+  std::string_view description;
+  bool is_required = false;
+  std::string_view valid_regex;
+  std::array<std::string_view, NumAllowedValues> allowed_values;
+  std::array<std::string_view, NumAllowedTypes> allowed_types;
```

Comment:
I think we could store these as actual types, instead of type name strings, and defer the string conversion to later. `utils/meta/type_list.h` could optionally be used for one kind of implementation.

Reply:
Since this was a list of strings before, as well, I would prefer to do this separately.  I have created a Jira for it: https://issues.apache.org/jira/browse/MINIFICPP-2164.